Settings Page

Daniel3886 opened the pull request:
Basic structure of Settings Page.

Softjey commented on client/src/components/layouts/HomeLayout.tsx:
```diff
@@ -17,7 +17,7 @@ const HomeLayout: React.FC<Props> = ({
   ...rest
 }) => {
-  const contentWidth = 1000;
+  const contentWidth = 100;
```
don'y change home layout

Softjey commented on client/src/pages/ChangePinCode.tsx:
all components have to be in components folder

Softjey commented on client/src/pages/ChangeTheme.tsx:
```diff
@@ -0,0 +1,74 @@
+      color: 'yellow',
+      transform: 'translateX(22px)',
+      '& .MuiSwitch-thumb:before': {
+        backgroundImage: `url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="20" width="20" viewBox="0 0 20 20"><path fill="${encodeURIComponent(
```
don't use text hardcoded icons, use mui icons components instead 
https://mui.com/material-ui/material-icons/

Softjey commented on client/src/pages/ChangeTheme.tsx:
```diff
@@ -0,0 +1,74 @@
+      top: 0,
+      backgroundRepeat: 'no-repeat',
+      backgroundPosition: 'center',
+      backgroundImage: `url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="20" width="20" viewBox="0 0 20 20"><path fill="${encodeURIComponent(
```
same here
https://mui.com/material-ui/material-icons/

Softjey commented on client/src/pages/SettingsPage.tsx:
```diff
@@ -0,0 +1,32 @@
+    <HomeLayout>
+      <Box sx={{ bgcolor: 'background.paper', color: 'text.primary', height: '100vh', p: 1 }}>
+        <List sx={{ width: '100%', maxWidth: 350, paddingTop: 0, bgcolor: 'background.paper' }}>
+          <Box sx={{ mb: 1 }}>
```
use flexbox instead of boxes with margin

Softjey commented on client/src/types/api/entities/user.ts:
don't chnage this file

Softjey commented on server/package-lock.json:
don't chnage this file

Daniel3886 commented:
Hello Softjey! I have reviewed your email and I have changed everything as
you requested. I couldn't quite understand what you meant by not changing a
file called "server/package-lock.json
<https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643557>".
The issue is that I never touched the file and changed anything there
If possible please provide more details about the issue
Thank you for understanding.
Daniel Czerczon

вс, 2 июн. 2024 г. в 21:59, Softjey ***@***.***>:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In client/src/components/layouts/HomeLayout.tsx
> <https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643054>:
>
> > @@ -17,7 +17,7 @@ const HomeLayout: React.FC<Props> = ({
>    centeredSx,
>    ...rest
>  }) => {
> -  const contentWidth = 1000;
> +  const contentWidth = 100;
>
> don'y change home layout
> ------------------------------
>
> On client/src/pages/ChangePinCode.tsx
> <https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643095>:
>
> all components have to be in components folder
> ------------------------------
>
> In client/src/pages/ChangeTheme.tsx
> <https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643377>:
>
> > +import Brightness7Icon from ***@***.***/icons-material/Brightness7';
> +import { styled } from ***@***.***/material/styles';
> +
> +const MaterialUISwitch = styled(Switch)(({ theme }) => ({
> +  width: 62,
> +  height: 34,
> +  padding: 7,
> +  '& .MuiSwitch-switchBase': {
> +    margin: 1,
> +    padding: 0,
> +    transform: 'translateX(6px)',
> +    '&.Mui-checked': {
> +      color: 'yellow',
> +      transform: 'translateX(22px)',
> +      '& .MuiSwitch-thumb:before': {
> +        backgroundImage: `url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="20" width="20" viewBox="0 0 20 20"><path fill="${encodeURIComponent(
>
> don't use text hardcoded icons, use mui icons components instead
> https://mui.com/material-ui/material-icons/
> ------------------------------
>
> In client/src/pages/ChangeTheme.tsx
> <https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643425>:
>
> > +    },
> +  },
> +  '& .MuiSwitch-thumb': {
> +    backgroundColor: theme.palette.mode === 'dark' ? '#003892' : '#001e3c',
> +    width: 32,
> +    height: 32,
> +    '&::before': {
> +      content: "''",
> +      position: 'absolute',
> +      width: '100%',
> +      height: '100%',
> +      left: 0,
> +      top: 0,
> +      backgroundRepeat: 'no-repeat',
> +      backgroundPosition: 'center',
> +      backgroundImage: `url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="20" width="20" viewBox="0 0 20 20"><path fill="${encodeURIComponent(
>
> same here
> https://mui.com/material-ui/material-icons/
> ------------------------------
>
> In client/src/pages/SettingsPage.tsx
> <https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643502>:
>
> > @@ -0,0 +1,32 @@
> +import React from 'react';
> +import { List, Box } from ***@***.***/material';
> +import HomeLayout from '../components/layouts/HomeLayout';
> +import ChangePinCode from './ChangePinCode';
> +import ChangeTheme from './ChangeTheme';
> +import ChangeUserInfo from './ChangeUserInfo';
> +import DeleteAccount from './DeleteAccount';
> +
> +const SettingsPage: React.FC = () => {
> +  return (
> +    <HomeLayout>
> +      <Box sx={{ bgcolor: 'background.paper', color: 'text.primary', height: '100vh', p: 1 }}>
> +        <List sx={{ width: '100%', maxWidth: 350, paddingTop: 0, bgcolor: 'background.paper' }}>
> +          <Box sx={{ mb: 1 }}>
>
> use flexbox instead of boxes with margin
> ------------------------------
>
> On client/src/types/api/entities/user.ts
> <https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643531>:
>
> don't chnage this file
> ------------------------------
>
> On server/package-lock.json
> <https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643557>:
>
> don't chnage this file
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Softjey/react_Examix/pull/47#pullrequestreview-2092598931>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A23XQ2F46HGQZC3ABETLRS3ZFN2SZAVCNFSM6AAAAABIUDGTRCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAOJSGU4TQOJTGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
